### PR TITLE
Add GCU test cases for k8s table

### PR DESCRIPTION
**Pull request by lixiaoyuner**
### Description of PR

Summary:
Fixes # (issue)
Testcase of k8s config for generic updater apply-patch

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [x] 202305
- [x] 202311

#### What is the motivation for this PR?
This PR is to verify the usage of 'config apply-patch' works for k8s config
#### How did you do it?
Add k8s config to dut and check if config changes as expected
#### How did you verify/test it?
Run test of sonic-mgmt/tests/generic_config_updater/test_kubernetes_config.py on KVM

### CODE REVIEW

File: tests/generic_config_updater/test_kubernetes_config.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            set(original_k8s_config) == set(current_k8s_config),
+            "k8s config are not suppose to change after test org: {}, cur: {}"
+            .format(original_k8s_config, current_k8s_config)
+        )
+
+    finally:
+        delete_checkpoint(duthost)
+
+
+def get_k8s_runningconfig(duthost):
```
Comment (wen587) [inline tests/generic_config_updater/test_kubernetes_config.py:204]: Any other check method other than running config?
The success of  apply-patch operation should guarantee the CONFIG_DB update. 
Is there a show command such as  `show k8s server`?

Comment (lixiaoyuner) [reply tests/generic_config_updater/test_kubernetes_config.py:210]: Thanks for this comment, have add 'show kube server' related validation.

**Review by wen587 — APPROVED** [review]
LGTM

File: tests/generic_config_updater/test_kubernetes_config.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+WRONGIPK8STABLE = ("", "true")
+
+
+# Succeed and fail flag
+SUCCEED = "SUCCEED"
+FAIL = "FAIL"
+
+
+test_data_1 = {
+    0: (K8SEMPTYTOHALFPATCH, K8SHALFCONFIG, EMPTYK8STABLE, SUCCEED),
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_kubernetes_config.py:164]: >0 [](http://example.com/codeflow?start=4&length=1)

Maintaining the index is not necessary. Could you remove?

Comment (lixiaoyuner) [reply tests/generic_config_updater/test_kubernetes_config.py]: Thanks for this comment, fixed.

**Review by qiluo-msft — APPROVED** [review]

**Comment by StormLiangMS** [comment]
hi @lixiaoyuner could you run test with 202305 image with this one? Would like to avoid regression for 202305 branch at this stage.

**Comment by lixiaoyuner** [comment]
> hi @lixiaoyuner could you run test with 202305 image with this one? Would like to avoid regression for 202305 branch at this stage.

Have tested with 202305 latest image, it's good to run this test. And also tested with 202311 latest image, it's good to run this test.

**Comment by anamehra** [comment]
Hi @lixiaoyuner , with this new testcase on a KUBERNETES enabled build, we observed following loganalyzer errors:
```
               Match Messages:
E               Apr 26 04:49:06.951797 bjw-can-8102-1 ERR ctrmgrd.py: Refer file /tmp/tmpj2csr7f7kube_hints_ for troubleshooting tips
E               
E               Apr 26 04:49:06.952134 bjw-can-8102-1 ERR ctrmgrd.py: Join failed: HTTPSConnectionPool(host='k8svip.ap.gbl', port=6443): Max retries exceeded with url: /api/v1/namespaces/default/configmaps/kube-root-ca.crt (Caused by NameResolutionError("&lt;urllib3.connection.HTTPSConnection object at 0x7fce9a825250&gt;: Failed to resolve 'k8svip.ap.gbl' ([Errno -3] Temporary failure in name resolution)"))
E               
E               Apr 26 04:49:46.964672 bjw-can-8102-1 ERR ctrmgrd.py: Refer file /tmp/tmp4r0omiobkube_hints_ for troubleshooting tips
E               
E               Apr 26 04:49:46.964855 bjw-can-8102-1 ERR ctrmgrd.py: Join failed: HTTPSConnectionPool(host='k8svip.ap.gbl', port=6443): Max retries exceeded with url: /api/v1/namespaces/default/configmaps/kube-root-ca.crt (Caused by NameResolutionError("&lt;urllib3.connection.HTTPSConnection object at 0x7fce9a825c70&gt;: Failed to resolve 'k8svip.ap.gbl' ([Errno -3] Temporary failure in name resolution)"))
E
```

I see the testcase using the hostname 'k8svip.ap.gbl'. What is the expectation with the test environment?

Thanks!

**Comment by lixiaoyuner** [comment]
> Hi @lixiaoyuner , with this new testcase on a KUBERNETES enabled build, we observed following loganalyzer errors:
> 
> ```
>                Match Messages:
> E               Apr 26 04:49:06.951797 bjw-can-8102-1 ERR ctrmgrd.py: Refer file /tmp/tmpj2csr7f7kube_hints_ for troubleshooting tips
> E               
> E               Apr 26 04:49:06.952134 bjw-can-8102-1 ERR ctrmgrd.py: Join failed: HTTPSConnectionPool(host='k8svip.ap.gbl', port=6443): Max retries exceeded with url: /api/v1/namespaces/default/configmaps/kube-root-ca.crt (Caused by NameResolutionError("&lt;urllib3.connection.HTTPSConnection object at 0x7fce9a825250&gt;: Failed to resolve 'k8svip.ap.gbl' ([Errno -3] Temporary failure in name resolution)"))
> E               
> E               Apr 26 04:49:46.964672 bjw-can-8102-1 ERR ctrmgrd.py: Refer file /tmp/tmp4r0omiobkube_hints_ for troubleshooting tips
> E               
> E               Apr 26 04:49:46.964855 bjw-can-8102-1 ERR ctrmgrd.py: Join failed: HTTPSConnectionPool(host='k8svip.ap.gbl', port=6443): Max retries exceeded with url: /api/v1/namespaces/default/configmaps/kube-root-ca.crt (Caused by NameResolutionError("&lt;urllib3.connection.HTTPSConnection object at 0x7fce9a825c70&gt;: Failed to resolve 'k8svip.ap.gbl' ([Errno -3] Temporary failure in name resolution)"))
> E
> ```
> 
> I see the testcase using the hostname 'k8svip.ap.gbl'. What is the expectation with the test environment?
> 
> Thanks!

@anamehra , actually it's a fake hostname, not a real endpoint. For this case, we just want to test the GCU command to update k8s table config. And there is PR to ignore the error, https://github.com/sonic-net/sonic-mgmt/pull/12849. Any more concerns about this?

**Pull request merged**